hawkular-command-gateway-api/src/main/resources/schema/A0uthMessage.schema.json

```diff
@@ -25,6 +49,27 @@
   "properties": {
     "authentication": {
       "$ref": "#/definitions/authentication"
+    },
+    "recipientType": {
+      "$ref": "#/definitions/recipientType"
+    },
+    "recipient": {
+      "description" : "feedId or uiId",
+      "type": "string"
+    },
+    "senderType": {
+      "$ref": "#/definitions/recipientType"
+    },
+    "sender": {
+      "description" : "feedId or uiId",
+      "type": "string"
+    },
+    "attachments": {
+      "description" : "A list of binary attachments if any.",
+      "type": "array",
+      "items" : {
+        "$ref": "#/definitions/attachment"
+      }
```

Comment:
> I don't like having this in an "AuthMessage" since this attachments thing doesn't have anything to do with authentication.

I basically agree. I was originally thinking of introducing some kind of an `Envelope` to comprise the authentication, routing info, attachments, payloadType and payload message itself. But I could not quickly find an elegant way to deserialize such messages.

I agree that what I added does not suit the `AuthMessage` name anymore and I am open for renaming it.

> Also, keep in mind that the cmdgw server will scrub the entire Authentication portion of the message as soon as the message is received (we actually null it out).

I know that we null the `authentication` field out, which makes no harm to these sibling fields, does it?